core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -387,28 +437,14 @@ protected void doRun() throws Exception {
         }
     }
 
-    public static class RetryOnPrimaryException extends ElasticsearchException {
-        public RetryOnPrimaryException(ShardId shardId, String msg) {
-            this(shardId, msg, null);
-        }
-
-        public RetryOnPrimaryException(ShardId shardId, String msg, Throwable cause) {
-            super(msg, cause);
-            setShard(shardId);
-        }
-
-        public RetryOnPrimaryException(StreamInput in) throws IOException {
-            super(in);
-        }
-    }
-
     /**
      * Responsible for routing and retrying failed operations on the primary.
-     * The actual primary operation is done in {@link PrimaryPhase} on the
+     * The actual primary operation is done in {@link ReplicatedOperation} on the
      * node with primary copy.
      *
      * Resolves index and shard id for the request before routing it to target node
      */
+    @SuppressWarnings("Duplicates")
```

Comment:
I suspect this is leftover from when you had multiple implementations in the PR but can it be removed now?